source/parsing/Preprocessor_macros.cpp

```diff
@@ -59,6 +59,17 @@ void Preprocessor::createBuiltInMacro(string_view name, int value, string_view v
 MacroActualArgumentListSyntax* Preprocessor::handleTopLevelMacro(Token directive) {
     auto macro = findMacro(directive);
     if (!macro.valid()) {
+        if (options.ignoreDirectives.find(directive.valueText().substr(1)) !=
+            options.ignoreDirectives.end()) {
+            SmallVectorSized<Trivia, 16> trivia;
+
+            trivia.append(createSimpleDirective(directive));
+            while (currentToken.kind != TokenKind::EndOfFile && peekSameLine())
+                consume();
+            trivia.appendRange(directive.trivia());
```

Comment:
No need to take the directive's trivia. Trivia needs to be listed in-order such that when you traverse tokens in-order and print them and their trivia you get back the original source file, so the ordering actually matters quite a bit.